src/csvReaderWrapper.js

```diff
@@ -19,23 +19,35 @@ export const parseCSV = (csvfile, download, useDefaultColumnDataType) => {
   });
 };
 
-const isCellValid = (cell) => {
-  return cell !== undefined && cell !== "";
-}
+const cleanData = (data) => {
+  var cleanedData = []
 
-const isRowValid = (row) => {
-  var cells = Object.values(row);
-  return cells.every(isCellValid)
-}
+  for (var row of data) {
+    var cleanedRow = tryGetCleanedRow(row);
+    if (cleanedRow !== null) {
+      cleanedData.push(cleanedRow);
+    }
+  }
 
-const cleanData = (data) => {
-  var cleanedData = data.filter(row => isRowValid(row));
   return cleanedData;
 }
 
-const countRemovedRows = (originalData, cleanedData) => {
-  var removedRowsCount = originalData.length - cleanedData.length;
-  store.dispatch(setRemovedRowsCount(removedRowsCount));
+const tryGetCleanedRow = (row) => {
+  for (var cellName in row) {
```

Comment:
is `cellName` the column heading? In other places where we're accessing the value of a specific cell it's something like `cellValue = row[column]` I wonder if we want to do access the contents via row and column here too for consistency?

Reply:
That makes more sense, thanks!